rest-server/src/models/job.js

```diff
@@ -112,16 +107,19 @@ class Job {
       .headers(launcherConfig.webserviceRequestHeaders)
       .end((requestRes) => {
         try {
+          if (typeof requestRes.body === 'undefined') {
+            next(null, new Error('Get job error: could not connect to launcher.'));
+          }
           const requestResJson =
             typeof requestRes.body === 'object' ?
               requestRes.body :
               JSON.parse(requestRes.body);
           if (requestRes.status === 200) {
             next(this.generateJobDetail(requestResJson), null);
           } else if (requestRes.status === 404) {
-            next(null, new Error('JobNotFound'));
+            next(null, new Error('Could not find job.'));
           } else {
-            next(null, new Error('InternalServerError'));
+            next(null, new Error('Get job error.'));
```

Comment:
>next [](start = 12, length = 4)

can the webportal know the status code in this case?

Reply:
yes, http status will be set in controllers if error is not null in callback